src/battle/item/dried_shroom/lib.c

```diff
@@ -0,0 +1,175 @@
+    s32 c = get_variable(script, *args++);
+    s32 d = get_variable(script, *args++);
+
+    func_80071090(0, a, b, c, d);
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus N(func_802A12FC_71CFDC)(ScriptInstance* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+
+    playerData->curHP++;
+    if (playerData->curHP > playerData->curMaxHP) {
+        playerData->curHP = playerData->curMaxHP;
+    }
+    return ApiStatus_DONE2;
+}
+
+Script N(UseItemWithEffect) = SCRIPT({
+    if (SI_VAR(1) == 0) {
+        UseCamPreset(69);
+        sleep 10;
+
+        PlaySoundAtActor(ActorID_PLAYER, 8333);
+        SetAnimation(ActorID_PLAYER, 0, PlayerAnim_GOT_ITEM);
+        GetActorPos(ActorID_PLAYER, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+        SI_VAR(0) += 18;
+        SetActorSpeed(ActorID_PLAYER, 4.0);
+        SetGoalPos(ActorID_PLAYER, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+        PlayerRunToGoal(0);
+
+        SI_VAR(1) += 45;
+        SI_VAR(3) = SI_VAR(1);
+        SI_VAR(3) += 10;
+        SI_VAR(3) += 2;
+        PlayEffect(51, 1, SI_VAR(0), SI_VAR(3), SI_VAR(2), 1.0, 30, 0, 0, 0, 0, 0, 0, 0);
+        MakeItemEntity(SI_VAR(10), SI_VAR(0), SI_VAR(1), SI_VAR(2), 1, 0);
+        SI_VAR(10) = SI_VAR(0);
+
+        N(GiveRefund)();
+        sleep SI_VAR(0);
+
+        sleep 15;
+
+        N(GiveRefundCleanup)();
+        RemoveItemEntity(SI_VAR(10));
+    } else {
+        GetActorPos(ActorID_PLAYER, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+        PlaySoundAtActor(ActorID_PLAYER, 8333);
+        SetAnimation(ActorID_PLAYER, 0, PlayerAnim_GOT_ITEM);
+        sleep 4;
+
+        SI_VAR(1) += 45;
+        SI_VAR(3) = SI_VAR(1);
+        SI_VAR(3) += 10;
+        SI_VAR(3) += 2;
+        PlayEffect(51, 1, SI_VAR(0), SI_VAR(3), SI_VAR(2), 1.0, 30, 0, 0, 0, 0, 0, 0, 0);
+        MakeItemEntity(SI_VAR(10), SI_VAR(0), SI_VAR(1), SI_VAR(2), 1, 0);
+        SI_VAR(10) = SI_VAR(0);
+
+        sleep 15;
+        RemoveItemEntity(SI_VAR(10));
+    }
+});
+
+Script N(UseItem) = SCRIPT({
+    UseCamPreset(19);
+    SetBattleCamTarget(0xFFFFFFAB, 1, 0);
+    SetBattleCamOffsetZ(41);
+    SetBattleCamZoom(248);
+    MoveBattleCamOver(30);
+    sleep 10;
+    SetAnimation(ActorID_PLAYER, 0, PlayerAnim_GOT_ITEM);
+    GetActorPos(ActorID_PLAYER, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    SI_VAR(1) += 45;
+    MakeItemEntity(SI_VAR(10), SI_VAR(0), SI_VAR(1), SI_VAR(2), 1, 0);
+    SI_VAR(14) = SI_VAR(0);
+    N(GiveRefund)();
+    sleep SI_VAR(0);
+    sleep 15;
+    N(GiveRefundCleanup)();
+    RemoveItemEntity(SI_VAR(14));
+});
+
+Script N(PlayerGoHome) = SCRIPT({
+    UseIdleAnimation(ActorID_PLAYER, 0);
+    SetGoalToHome(0);
+    SetActorSpeed(ActorID_PLAYER, 8.0);
+    SetAnimation(ActorID_PLAYER, 0, PlayerAnim_RUNNING);
+    PlayerRunToGoal(0);
+    SetAnimation(ActorID_PLAYER, 0, PlayerAnim_2);
+    UseIdleAnimation(ActorID_PLAYER, 1);
+});
+
+Script N(EatItem) = SCRIPT({
+    spawn {
+        loop 4 {
+            PlaySoundAtActor(ActorID_PLAYER, 8341);
```

Comment:
We probably want the second argument to instances of PlaySoundAtActor to be in hex, assuming it's a sound ID, and we might even have the define for it